test/DurableTaskEndToEndTests.cs

```diff
@@ -663,6 +688,40 @@ public DurableTaskEndToEndTests(ITestOutputHelper output)
             await this.SubOrchestration_ComplexType_Main_Logic(true, true);
         }
 
+        /// <summary>
+        /// End-to-end test which validates a sub-orchestrator function have assigned corrent value for <see cref="DurableOrchestrationContext.ParentInstanceId"/>.
+        /// </summary>
+        [Fact]
+        public async Task SubOrchestration_Has_Valid_ParentInstanceId_Assigned()
+        {
+            const string TaskHub = nameof(this.SubOrchestration_ComplexType);

```

Comment:
The convention for task hub names in tests is to use the name of the test. In this case `nameof(this.SubOrchestration_Has_Valid_ParentInstanceId_Assigned)`